### PR TITLE
[W5][T08-3]Sean Yap

### PR DESCRIPTION
**Pull request by seanieyap**
Added sort functionality that sorts all the people in the address book into ascending order according to their full name. Only in ascending order for now.

### CODE REVIEW

**Comment by marcus-pzj** [comment]
Overall a good working implementation of the added functionality with correct coding style!
Great update to I/O & JUnit Tests!
Just some minor comments below:

https://github.com/nusCS2113-AY1819S2/addressbook-level2/blob/e95fec5d9eeef3f99259ea46b369b531efe21ba5/src/seedu/addressbook/commands/SortCommand.java#L1-L8 

Consider updating the comments to indicate the new behavior that you have incorporated in this class.

https://github.com/nusCS2113-AY1819S2/addressbook-level2/blob/e95fec5d9eeef3f99259ea46b369b531efe21ba5/test/java/seedu/addressbook/data/AddressBookTest.java#L126-L132

I noticed that you implemented the sort() test in Addressbook Class. Since sort is a command, maybe it is better to create a SortCommandTest under commands instead? Nevertheless, this test still suffices in checking for functionality of the sort function implemented.

**Comment by seanieyap** [comment]
> 
> 
> Overall a good working implementation of the added functionality with correct coding style!
> Great update to I/O & JUnit Tests!
> Just some minor comments below:
> 
> [addressbook-level2/src/seedu/addressbook/commands/SortCommand.java](https://github.com/nusCS2113-AY1819S2/addressbook-level2/blob/e95fec5d9eeef3f99259ea46b369b531efe21ba5/src/seedu/addressbook/commands/SortCommand.java#L1-L8)
> 
> Lines 1 to 8 in [e95fec5](/nusCS2113-AY1819S2/addressbook-level2/commit/e95fec5d9eeef3f99259ea46b369b531efe21ba5)
>  package seedu.addressbook.commands; 
>   
>   
>  /** 
>   * Lists all persons in the address book to the user. 
>   */ 
>  public class SortCommand extends Command { 
>   
> 
> Consider updating the comments to indicate the new behavior that you have incorporated in this class.

Oh yes, thanks for informing this, a clumsy mistake on my part ! This has been edited in the latest commit.
 
> [addressbook-level2/test/java/seedu/addressbook/data/AddressBookTest.java](https://github.com/nusCS2113-AY1819S2/addressbook-level2/blob/e95fec5d9eeef3f99259ea46b369b531efe21ba5/test/java/seedu/addressbook/data/AddressBookTest.java#L126-L132)
> 
> Lines 126 to 132 in [e95fec5](/nusCS2113-AY1819S2/addressbook-level2/commit/e95fec5d9eeef3f99259ea46b369b531efe21ba5)
>  @Test 
>  public void sort() throws Exception { 
>      UniquePersonList unsortedPersons = defaultAddressBook.getAllPersons(); 
>      UniquePersonList personsToCheck = new UniquePersonList(aliceBetsy, bobChaplin, charlieDouglas, davidElliot); 
>      unsortedPersons.sort(); 
>      assertTrue(isIdentical(unsortedPersons, personsToCheck)); 
>  } 
> 
> I noticed that you implemented the sort() test in Addressbook Class. Since sort is a command, maybe it is better to create a SortCommandTest under commands instead? Nevertheless, this test still suffices in checking for functionality of the sort function implemented.

I chose not to do this as similarly with the list command, these commands are not the same as with the others, there is no change in output and no exception/error messages thrown regardless of the state of the program. But it is true that it is still a command on its own. This could be done in the future.
the list command was too not implemented the same way as with other commands. 

I've also realized that i have forgotten to edit the help command, so that has been updated as well. Cheers

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -151,6 +151,11 @@ Views all details of the 2nd person in the address book.
 `viewall 1` +
 Views all details of the 1st person in the results of the `find` command.
 
+== Sorting all entries : `sort`
+
+Sort sorts all entries in the address book by name in ascending order. +
+Format: `sort`
+
```
Comment (0WN463) [inline docs/UserGuide.adoc:158]: Interesting choice of words

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n" 
+            + "Sort all persons in the address book in ascending order.\n"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "Address book has been sorted by ascending order!";
+
```
Comment (0WN463) [inline src/seedu/addressbook/commands/SortCommand.java:16]: Very very small nitpick. Ascending order does not have meaning when referring to names. If you want to be accurate, it should be lexicographical order or something.

File: test/java/seedu/addressbook/data/AddressBookTest.java [hunk 3]
```diff
@@ -120,7 +120,15 @@ public void removePerson_personNotExists_throwsPersonNotFoundException() throws
...
     }
-
+    
+    @Test
+    public void sort() throws Exception {
+        UniquePersonList unsortedPersons = new UniquePersonList(davidElliot, charlieDouglas, bobChaplin, aliceBetsy);
+        UniquePersonList sortedPersons = new UniquePersonList(aliceBetsy, bobChaplin, charlieDouglas, davidElliot);
+        unsortedPersons.sort();
+        assertTrue(isIdentical(unsortedPersons, sortedPersons));
+    }
```
Comment (0WN463) [inline test/java/seedu/addressbook/data/AddressBookTest.java:130]: Good use of JUnit testing

File: test/expected.txt [hunk 4]
```diff
@@ -198,6 +201,21 @@
 || Enter command: || [Command entered:  viewall 5]
 || Viewing person: Esther Potato Phone: 555555 Email: esther@not.a.real.potato Address: (private) 555, epsilon street Tags: [tubers][starchy]
 || ===================================================
+|| Enter command: || [Command entered:  sort a]
+|| Address book has been sorted by ascending order!
+|| ===================================================
```
Comment (0WN463) [inline test/expected.txt:206]: Interesting that you chose to catch for trailing arguments (good design to do so), however, your feedback message still indicates a success, which muddles the true outcome

**Review by 0WN463 — COMMENTED** [review]
Overall, good attempt, especially with the updating of I/O testing and JUnit testing

**Comment by 0WN463** [comment]
See my comments for suggestions. Please close this PR after reading comments.

**Pull request closed without merging**